client/login/redirect-logged-in/index.web.js

```diff
@@ -1,14 +1,38 @@
 import safeProtocolUrl from 'calypso/lib/safe-protocol-url';
 import { isUserLoggedIn } from 'calypso/state/current-user/selectors';
 
+/**
+ * For this context, we consider external URLs that are NOT:
+ * - Relative paths (`/test`)
+ * - Absolute URLs on https://wordpress.com/*
+ * @param {string} url URL to check
+ * @returns {boolean}
+ */
+function isExternalUrl( url ) {
+	if ( url.startsWith( '/' ) ) {
+		return false;
+	}
+
+	try {
+		const urlObject = new URL( url );
+		if ( urlObject.hostname === 'wordpress.com' && urlObject.protocol === 'https:' ) {
```

Comment:
If I am not mistaken you used the try-catch for cases that `url` cannot be constructed. In this case, I think moving the condition after the try-catch would make the code more readable.

Reply:
We'd have to declare `urlObject` as a variable, and outside of the `try/catch` though. I think I prefer it as it is.

Reply:
Let me know if you have strong feelings about it though

Reply:
That's fair!